lib/metrics/metrics-adapter.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE
+
+import {getMessage} from '../utils/messages';
+import {MetricsResults, Timing} from '../../types/types';
+import METRICS from './metrics';
+
+const checkAudits = (audits: Record<string, LH.Audit.Result>) => Object.keys(audits).forEach(key => {
+  const errorMessage = audits[key].errorMessage;
+  if (errorMessage)
+    console.log(`${errorMessage} Audit key: ${key}`);
+});
+
+const getMetricTitle = (metricId) => {
+  try {
+    return getMessage(metricId);
+  } catch (e) {
+    return '';
+  }
+};
+
+export const adaptMetricsData = (res: LH.Result): MetricsResults => {
+  const audits:Record<string, LH.Audit.Result>  = res.audits;
+
+  checkAudits(audits);
+
+  const colorP0 = 'yellow';
+  const colorP2 = 'green';
+  const colorVisual = 'blue';
+
+  const timings: Timing[] = [];
+
+  Object.keys(audits).forEach(metricKey => {
+    const metric = audits[metricKey];
```

Comment:
the `audits['metrics']` audit contains everything i think you want. up to you, but it would be slightly cleaner to pull from just that.

https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/audits/metrics.js#L96-L138

btw VizCommplete is in there as `observedLastVisualChange`. however that's **not** adjusted if run with simulated throttling aka lantern.

Reply:
So to clarify, if we debug LH method `_runAudit` in runner.js, while we call 
`const product = await audit.audit(artifacts, {options: auditOptions, settings: settings});`

`artifacts` has all stuff needed for `Metrics.audit` method, 
but if we will look at artifacts which LH provide after run we will see mostly the same object except all needed methods like `requestCriticalRequestChains`, `requestEstimatedInputLatency`, `requestFirstCPUIdle` and many others. If I'm not wrong it's ComputedArtifacts.

That's why call `Metrics.audit` will crash on lines https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/audits/metrics.js#L34-L41

Maybe it's a bug and LH has also provide artifacts with all necessary methods, so they can be used in `Metrics.audit` ?

Reply:
Hmmmmmmmm

This is getting into some Lighthouse internals stuff that I don't think pwmetrics needs to worry about, but maybe I'm missing something. I'm saying the config will be have `onlyAudits: ['metrics']`. No more filtering to the metric group ids.. Just pulling this singular audit. By including it, it should compute all the metrics.

`adaptMetricsData` still gets passed `data.lhr`. But you'll pull numbers out like:

```js
lhr.audits.metrics.details[metricName]
lhr.audits.metrics.details[`${metricName}Ts`]
```

Reply:
sorry for the delay,
yup it looks more reasonable:) I just got confused and dug in wrong diraction 💫 
hope to get back to it later today)